app/controllers/products_controller.rb

```diff
@@ -0,0 +1,45 @@
+class ProductsController < ApplicationController
+  before_action :set_nav
+  before_action :set_product, except: [:new, :create]
```

Comment:
Either of those is what I was thinking. I think this particular instance is totally cool to leave as is. I think I was meaning that comment as a more general observation. If you had a handful of these kinds of `before_action` calls happening, that can get tricky to follow. When you start dealing with bigger, older projects, it's important to make sure you'll be able to figure out what's going on  in a few weeks' time.